src/cmap/connect.ts

```diff
@@ -307,6 +307,10 @@ function parseConnectOptions(options: ConnectionOptions): SocketConnectOpts {
       (result as Document)[name] = options[name];
     }
   }
+  if (result.keepAliveInitialDelay == null) {
+    result.keepAliveInitialDelay = 120000;
+  }
+  result.keepAlive = true;
```

Comment:
Can't comment on the line but can we also remove the `socket.setNoDelay(noDelay);` line in favor of passing it in on the options (which it already is)

It appears to not be supported on the client options so this should always be true, we can give it the same treatment as `keepAlive = true` (unconditional assignment)

Reply:
I actually have a test now with it in the client options and verify it gets passed through. (https://github.com/mongodb/node-mongodb-native/pull/4510/files#diff-5167348e5ef5edeae43b202c9f7b7c34c55cc63057ae7d311db89fef8a26c13fR271)

Reply:
ah neat even simple, so we can just not call the `setNoDelay` method because it is redundant right?